.gitignore

```diff
@@ -1,24 +1,20 @@
-*.iml
-*.lock
-*.komodoproject
```

Comment:
This is an example of what I mean by odd

Reply:
Yea. I have no idea what that is.

Reply:
So I don't know how strongly you feel about this. The main reason why I did this change in the first place is as you can tell there is a lot of weird stuff in this `gitignore` and it took me a few minutes to just create a nice clean one.

I can go through the `.gitignore` to just remove the really weird stuff but tbh almost every entry that isnt editor/IDE related seems wrong one way or another. I get your point about people not doing the proper thing and using a global gitignore but in my view I don't think should block the PR. Assuming we merge this PR, if someone in the future submits a PR and then complains that their files are being committed and add an entry to the `.gitignore` you can always accept it.

Alternately I can just add the various IDE/OS specific entries we are talking about but to me this is unnecessary.